### ISSUE TITLE
build/doc: make it easier to identify missing dependencies on Ubuntu 18.04.1 LTS

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-sched#396]
While testing #394 on a Ubuntu 18.04.1 LTS system, I ran into a few snags that could have been avoided with better documentation and/or configure logic.

1) libxml2-dev - this is caught by configure as it ought to be, but should be added to README.md

2) boost - docs state only that libboost-dev is required, but with only that installed on this Ubuntu release, one gets various unhelpful configure errors like
```
checking for boostlib >= 1.53.0 (105300)... yes
checking for buggy boostlib versions (105400-105800)... no
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
configure: error: Could not link against boost_system !
```
until all the required boost sub packages are installed:  libboost-system-dev, libboost-thread-dev, libboost-graph-dev, libboost-filesystem-dev, libboost-regex-dev

3) python-yaml - flux-sched configures and builds ok, but fails some `make check` tests.   Running one failing test standalone with `--verbose` yields e.g.
```
expecting success:
    flux resource match allocate ${jobspec} &&
    flux resource match allocate ${jobspec} &&
    flux resource match allocate ${jobspec} &&
    flux resource match allocate ${jobspec}

Traceback (most recent call last):
  File "/home/garlick/proj/flux-sched/t/scripts/flux-resource", line 5, in <module>
    import yaml
ImportError: No module named yaml
not ok 2 - match-allocate works with a 1-node, 1-socket jobspec
```
Looking at the `flux-resource` script, I find it's requiring the python yaml module.

`make check` runs clean on this system after pulling in all the required dependencies.

**Comment by SteVwonder** [comment]
Nice catch, @garlick!

If the part about boost is an easy fix, maybe it can get added to #394 before it's merged.

For the yaml piece, @trws wrote [a nice m4 macro](https://github.com/flux-framework/flux-core/blob/master/config/am_check_pymod.m4) for detecting python module versions.  We should pull that into sched and check for the python yaml module.

**Comment by dongahn** [comment]
Thanks for referring to this issue as part of PR #394. 

The fixes should be easy enough to add to PR #394. 

BTW, I didn't know if `libboost-thread-dev` was required. Probably, a sub requirement of one of the top level dependencies. Will add all of them to the README.md in any case.

For Python yaml, I will pull in the macro. We probably don't have python listed in README, so that needs to be updated there as well.
